buildSrc/src/main/groovy/local.python.container-service-conventions.gradle

```diff
@@ -17,9 +35,115 @@ docker {
+}
+
+def dockerStart = tasks.register('dockerStart', DefaultTask) {
+    group = "Docker Run"
+    description = "Remove unused containers, build image, run container"
+    dependsOn tasks.named("dockerPrune")
+    dependsOn tasks.named("docker")
+    dependsOn tasks.named("dockerRun")
+}
+
+def dockerPruneContainer = tasks.register('dockerPruneContainer', Exec) {
+    group = "Docker"
+    description = "Remove unused containers"
+    executable "docker"
+    args "container", "prune", "-f"
+}
+
+def dockerPruneImage = tasks.register('dockerPruneImage', Exec) {
+    group = "Docker"
+    description = "Remove unused images"
+    executable "docker"
+    args "image", "prune", "-f"
+}
+
+def dockerPruneVolume = tasks.register('dockerPruneVolume', Exec) {
+    group = "Docker"
+    description = "Remove unused volumes"
+    executable "docker"
+    args "volume", "prune", "-f"
+}
+
+def dockerPrune = tasks.register('dockerPrune', DefaultTask) {
+    group = "Docker"
+    description = "Stop container, then remove unused containers and images"
+    mustRunAfter tasks.named('dockerStop')
+    mustRunAfter tasks.named('dockerRemoveContainer')
+    dependsOn dockerPruneContainer
+    dependsOn dockerPruneImage
+}
+
+def dcPrune = tasks.register('dcPrune', DefaultTask) {
+    group = "Docker Container"
+    description = "Stop containers, then remove unused containers and images"
+    mustRunAfter tasks.named('dockerComposeDown')
+    dependsOn dockerPruneContainer
+    dependsOn dockerPruneImage
+}
+
+def dcPruneVolume = tasks.register('dcPruneVolume', DefaultTask) {
+    group = "Docker Container"
+    description = "Stop containers, then remove unused volumes"
+    mustRunAfter tasks.named('dockerComposeDown')
+    dependsOn dockerPruneVolume
+}
+
+def lintDockerfile = tasks.register('lintDockerfile', DefaultTask) {
+    group = JavaBasePlugin.VERIFICATION_GROUP
+    description = "Use linter (default hadolint) to perform static analysis on Dockerfile"
+    ext.binary = "hadolint"
+    ext.targets = [ "../Dockerfile" ]
+    ext.taskTimeout = 10000L
+    def result = 0
+    def sout = new StringBuilder()
+    def serr = new StringBuilder()
+    doLast {
+        ext.targets.each { f ->
+            def cmdLine = "${ext.binary} ${f}"
+            def proc = cmdLine.execute(null, project.projectDir)
+            proc.consumeProcessOutput(sout, serr)
+            proc.waitForOrKill(ext.taskTimeout)
+            result |= proc.exitValue()
+        }
+        if (result != 0 && serr) {
+            ant.fail(serr) }
+        else if (result != 0) {
+            ant.fail(sout)
+        }
+        return result
+    }
+}
+
+tasks.named("dockerRemoveContainer").configure {
+    mustRunAfter tasks.named('dockerStop')
+}
+
+tasks.named("dockerComposeUp").configure {
+    dependsOn tasks.named("docker")
+}
+
+tasks.named("dockerRun").configure {
+    dependsOn tasks.named("docker")
+}
+
+tasks.named("check").configure {
+    dependsOn lintDockerfile
+}
+
```

Comment:
Since almost all of this is copied from `local.java.container-conventions.gradle`, let's create a new `local.docker.container-conventions.gradle` and add it (`id 'local.docker.container-conventions'`) to the top of this file and `local.java.container-conventions.gradle`.
One difference is:
```
dockerCompose {
    dockerComposeFile 'docker-compose.yml'
}
```
which you can keep in this file.